src/qml/mixxxscreen.h

```diff
@@ -0,0 +1,105 @@
+//
+// Created by augier on 15/07/24.
+//
+
+#ifndef MIXXX_MIXXXSCREEN_H
+#define MIXXX_MIXXXSCREEN_H
+
+#include <QtQml/qqmlregistration.h>
+
+#include <QImage>
+#include <QObject>
+#include <QQuickItem>
+#include <QSize>
+
+#include "mixxxcontroller.h"
+
+namespace mixxx {
+namespace qml {
+
+class MixxxScreen : public QObject, public QQmlParserStatus {
+    Q_OBJECT
+    QML_ELEMENT
+    Q_INTERFACES(QQmlParserStatus)
+
+    Q_PROPERTY(QString screenId MEMBER m_screenId REQUIRED)
+    Q_PROPERTY(int width READ width WRITE setWidth)
+    Q_PROPERTY(int height READ height WRITE setHeight)
+    Q_PROPERTY(uint targetFps MEMBER m_targetFps)
+    Q_PROPERTY(uint msaa MEMBER m_msaa)
+    Q_PROPERTY(uint splashOff READ splashOff WRITE setSplashOff)
+    Q_PROPERTY(QImage::Format pixelType MEMBER m_pixelType)
+    Q_PROPERTY(ColorEndian endian MEMBER m_endian)
+    Q_PROPERTY(bool reversedColor MEMBER m_reversedColor)
+    Q_PROPERTY(bool rawData MEMBER m_rawData)
+    Q_PROPERTY(QJSValue transformFrame READ jsTransformFrame WRITE
+                    setJsTransformFrame NOTIFY jsTransformFrameChanged)
+
+    Q_PROPERTY(QQuickItem* item MEMBER m_item)
+
+    Q_CLASSINFO("DefaultProperty", "item")
+
+  public:
+    enum class ColorEndian {
+        Big = static_cast<int>(std::endian::big),
+        Little = static_cast<int>(std::endian::little),
+    };
+    Q_ENUM(ColorEndian)
+
+    explicit MixxxScreen(MixxxController* parent = nullptr)
```

Comment:
So I did the test and `parent` here is `nullptr`. How can I force a QML component to be instanciated inside a `MixxxController`.